### ISSUE TITLE
use official upstream Atomic Desktop images (once they support bootc) insted of current unofficial

### ISSUE THREAD

**Issue by bsherman** [ublue-os/main#669]
This is pending Fedora F41 turning on bootupd for BIOS systems

Also these changes may be related:
https://pagure.io/fedora-infra/ansible/pull-request/2337
https://pagure.io/releng/issue/12142

**Comment by castrojo** [comment]
The latest lazurite-main has 21 pulls: https://github.com/ublue-os/main/pkgs/container/lazurite-main/versions

The latest vauxite-main has 92 pulls: https://github.com/ublue-os/main/pkgs/container/vauxite-main

lazurite is probably scanner noise, let's nuke it, but there's probably a handful of xfce folks on those images. Timothee [says this](https://discussion.fedoraproject.org/t/whats-holding-back-fedora-to-create-an-offical-xfce-atomic-spin/115566/2) on an official xfce image, so we can probably point to people there if they want to continue getting support.

**Comment by alexsch01** [comment]
Curious, I thought the bootc images don't allow layering, is that still true?

**Comment by bsherman** [comment]
bootc is the future of Fedora Atomic Desktops and CoreOS, so it's our future. 

Local layering has never been a goal of Universal Blue as I've understood it. The vision has always been a future where that's not required. Local layering has been a painful occasional workaround. 

All said, I should answer your question 😃

No bootc does not support local layering, but we will still have a container native rpm-ostree backend, so users wanting to local layer can still use rpm-ostree for that purpose. 

Switching our source images is an important step but will not automatically change everything to upgrade via bootc, which is where local layers would be dropped.

**Comment by castrojo** [comment]
bootc tells the user what to do: 

```
# bootc update
ERROR Upgrading: Deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications.
```

**Comment by tulilirockz** [comment]
I believe this can be worked on now!  - Literally everything mentioned on the https://gitlab.com/fedora/ostree/sig/-/issues/1 URL seems to be fixed/merged :)

**Comment by gerblesh** [comment]
As per issue: https://gitlab.com/fedora/ostree/sig/-/issues/48 and https://pagure.io/releng/issue/12142
bootc, bootupd, and dnf5 should now all be working on the official images. I'm not sure if we want to move right away, if we were to switch, we would likely have to only move images that are F41+

**Comment by castrojo** [comment]
Not sure if this should be a separate issue, but I think we can drastically cut down on the amount of images we generate by taking advantage of the bootc/dnf work more directly.

Here are all the groups in there so far: https://gist.github.com/castrojo/6dc7ca127b1976052dfb03d8376fe7d7

I was able to add COSMIC and it's apps to my custom image with one line, check it out: https://github.com/castrojo/achillobator/blob/main/build.sh

I could do the same with budgie if I wanted. In the past due to rpm-ostree's limitations we ended up trying to make finished desktops by doing the individual packages and then adding in all the QoL. Now with dnf we can just use the existing Fedora groups, which means a custom image can look like:

    FROM ghcr.io/ublue-os/base-main        
    RUN dnf5 install $desktop-group apps-I-want

and you're done. Then instead of trying to make an nvidia variant for each desktop people ask for they'd derive from `base-nvidia` instead. There'd be no need to explode the matrix for every time someone wants a new window manager, they can just use the existing dnf groups, which are the exact same ones workstation uses.

Then bluebuild can do a `desktop: budgie` module or whatever. 

It also means that when we deprecate an image, say XFCE, the people who want it have a path forward. They go from:

    FROM ghcr.io/ublue-os/vauxite-main
    RUN $all_my_stuff

to 

    FROM ghcr.io/ublue-os/base-main
    RUN dnf5 install xfce-group-thing
    RUN $all_my_stuff

**Comment by tulilirockz** [comment]
That will require a tiny bit of curating since dnf5 does not install a ton of dependencies that rpm-ostree installs automatically, like, `gnome-bluetooth`, `gnome-remote-desktop`, and others for some reason! - But this seems like the best way going forward for SURE, it will let people more easily customize their own images, itll make it so they can have super stripped down things, will make the maintenance burden a lot lower on the `main` image, AND it also will make it a ton easier to change bases if someone wants to

**Comment by tulilirockz** [comment]
and it also deincentivizes `main` image usage by literally not having them!!! GREAT idea to be completely honest

**Comment by KyleGospo** [comment]
> Not sure if this should be a separate issue, but I think we can drastically cut down on the amount of images we generate by taking advantage of the bootc/dnf work more directly.
> 
> Here are all the groups in there so far: https://gist.github.com/castrojo/6dc7ca127b1976052dfb03d8376fe7d7
> 
> I was able to add COSMIC and it's apps to my custom image with one line, check it out: https://github.com/castrojo/achillobator/blob/main/build.sh
> 
> I could do the same with budgie if I wanted. In the past due to rpm-ostree's limitations we ended up trying to make finished desktops by doing the individual packages and then adding in all the QoL. Now with dnf we can just use the existing Fedora groups, which means a custom image can look like:
> 
> ```
> FROM ghcr.io/ublue-os/base-main        
> RUN dnf5 install $desktop-group apps-I-want
> ```
> 

This is a -1 from me, group install is cool but Spins do a lot more than a single group install command can get you, and we'd lose out on all of that work.

**Comment by noelmiller** [comment]
> Not sure if this should be a separate issue, but I think we can drastically cut down on the amount of images we generate by taking advantage of the bootc/dnf work more directly.
> 
> Here are all the groups in there so far: https://gist.github.com/castrojo/6dc7ca127b1976052dfb03d8376fe7d7
> 
> I was able to add COSMIC and it's apps to my custom image with one line, check it out: https://github.com/castrojo/achillobator/blob/main/build.sh
> 
> I could do the same with budgie if I wanted. In the past due to rpm-ostree's limitations we ended up trying to make finished desktops by doing the individual packages and then adding in all the QoL. Now with dnf we can just use the existing Fedora groups, which means a custom image can look like:
> 
> ```
> FROM ghcr.io/ublue-os/base-main        
> RUN dnf5 install $desktop-group apps-I-want
> ```
> 
> and you're done. Then instead of trying to make an nvidia variant for each desktop people ask for they'd derive from `base-nvidia` instead. There'd be no need to explode the matrix for every time someone wants a new window manager, they can just use the existing dnf groups, which are the exact same ones workstation uses.
> 
> Then bluebuild can do a `desktop: budgie` module or whatever.
> 
> It also means that when we deprecate an image, say XFCE, the people who want it have a path forward. They go from:
> 
> ```
> FROM ghcr.io/ublue-os/vauxite-main
> RUN $all_my_stuff
> ```
> 
> to
> 
> ```
> FROM ghcr.io/ublue-os/base-main
> RUN dnf5 install xfce-group-thing
> RUN $all_my_stuff
> ```
> 
> Also, if Bazzite wants to do Budgie and COSMIC desktops they can just install those groups right in their repo and build it there instead of us having to amplify the matrix for every desktop.

Yeah, I agree with Kyle here. If we wanted to get rid of less used ones like vauxite and others, I would be in favor of that first before redoing all the work that Timothee and the spins are doing for us.

**Comment by noelmiller** [comment]
Also, probably separate issue, but we could look at starting to migrate to using DNF5 rather than using rpm-ostree.

**Comment by tulilirockz** [comment]
> > Not sure if this should be a separate issue, but I think we can drastically cut down on the amount of images we generate by taking advantage of the bootc/dnf work more directly.
> > Here are all the groups in there so far: https://gist.github.com/castrojo/6dc7ca127b1976052dfb03d8376fe7d7
> > I was able to add COSMIC and it's apps to my custom image with one line, check it out: https://github.com/castrojo/achillobator/blob/main/build.sh
> > I could do the same with budgie if I wanted. In the past due to rpm-ostree's limitations we ended up trying to make finished desktops by doing the individual packages and then adding in all the QoL. Now with dnf we can just use the existing Fedora groups, which means a custom image can look like:
> > ```
> > FROM ghcr.io/ublue-os/base-main        
> > RUN dnf5 install $desktop-group apps-I-want
> > ```
> 
> This is a -1 from me, group install is cool but Spins do a lot more than a single group install command can get you, and we'd lose out on all of that work.

I just checked out the official fedora silverblue yaml files they use to build and we would literally just be losing out on this:
![Image](https://github.com/user-attachments/assets/2ade6bad-603d-4021-b93f-5ba447afdac2)

They literally just use the DNF groups for the official images anyway
https://pagure.io/workstation-ostree-config/blob/main/f/comps-sync.py

**Comment by castrojo** [comment]
We can probably keep silverblue and kinoite and yeet the rest since those are the ones we're using for the downstreams anyway.

**Comment by averms** [comment]
> They literally just use the DNF groups for the official images anyway

That's not quite true. There is a lot of manual exclusion of packages going on in https://pagure.io/workstation-ostree-config/blob/main/f/comps-sync-exclude-list.yml

**Comment by travier** [comment]
> This is a -1 from me, group install is cool but Spins do a lot more than a single group install command can get you, and we'd lose out on all of that work.

Exactly, if you rebase the Universal Blue images as "dnf install on top of a bootc base image" then you will loose all the work that goes into curating what is or is not included in Silverblue, Kinoite, etc. which is only in the manifests for those right now.

---

I don't recommend rebasing on top of the "official" Fedora images yet.

The main issue is: https://pagure.io/cloud-image-uploader/issue/37. Sure, things should work well in general because Fedora is reasonably stable, until it doesn't and then you can not compare with previous images.

You also can not use them to create true container native workflows, pinning the base image version and bumping via depdendabot or similar.

The images are also not used anywhere in Fedora right now, so issues with them are not noticed, they are not being tested, they are not release blocker, fixes happen on a best effort basis, etc. It took me a month to notice that they were out of date and file a ticket.

The images are also not signed, and it's unlikely that they will ever get signed until we move to building them via Konflux.
I've added cosign signatures for the unofficial ones so that we can make progress on development for the graphical interface to surface the signing status.

**Comment by p5** [comment]
Thanks for the info!

> Exactly, if you rebase the Universal Blue images as "dnf install on top of a bootc base image" then you will loose all the work that goes into curating what is or is not included in Silverblue, Kinoite, etc. which is only in the manifests for those right now.

Completely get where you're coming from.

We're pretty sure our plan will be to follow upstream Fedora's lead on this, and build only the "official" atomic spins, plus one DE-less image.  I.e. those listed below:

![Image](https://github.com/user-attachments/assets/a21a2e13-e412-47ac-815b-ad8b47d1d64f)

We don't want to be building images when there are only a handful of users, since it's just not maintainable for us.  We're maxing out on GitHub Actions runners as-is, and we're looking to build ARM variants in the future.

The "unofficial" spins (i.e. the DEs not listed in the above image) I assume aren't going to be release-blocking anyway, so users should expect a lesser experience with those.

---

I'm going to split these into two distinct issues - those being switching to the quay.io/fedora images, and deprecating non-official images.
Deprecating non-official spins can be done for F42+, and we'll play it by ear as to when we can switch over to the official Quay registry.

**Comment by travier** [comment]
Still not ready, sorry. Hoping the Forgejo migration or a potential move to Konflux might help us.

**Comment by LorbusChris** [comment]
ref:
- https://forge.fedoraproject.org/atomic-desktops/config/pulls/764
- https://forge.fedoraproject.org/atomic-desktops/config/pulls/783
